src/core/network.js

```diff
@@ -115,20 +133,17 @@ var NetworkManager = (function NetworkManagerClosure() {
         pendingRequest.expectedStatus = 200;
       }
 
-      if (args.onProgressiveData) {
-        // Some legacy browsers might throw an exception.
-        try {
-          xhr.responseType = 'moz-chunked-arraybuffer';
-        } catch(e) {}
-        if (xhr.responseType === 'moz-chunked-arraybuffer') {
-          pendingRequest.onProgressiveData = args.onProgressiveData;
-          pendingRequest.mozChunked = true;
-        } else {
-          xhr.responseType = 'arraybuffer';
-        }
+//#if (GENERIC || FIREFOX || MOZCENTRAL)
+      if (args.onProgressiveData && supportsMozChunked) {
+        xhr.responseType = 'moz-chunked-arraybuffer';
+        pendingRequest.onProgressiveData = args.onProgressiveData;
+        pendingRequest.mozChunked = true;
       } else {
         xhr.responseType = 'arraybuffer';
       }
+//#else
```

Comment:
That certainly makes sense, and I've also understood it in that way.

By "preprocessor magic", I meant https://github.com/mozilla/pdf.js/pull/5531#discussion-diff-31866452. Assuming that setting `.responseType` multiple times is inexpensive (which probably is), the complete code (with minimal use of the boolean) could be:

```js
      xhr.responseType = 'arraybuffer';
//#if (MOZCENTRAL || FIREFOX)
//    if (args.onProgressiveData) {
//#endif
//#if !(MOZCENTRAL || FIREFOX || CHROMIUM)
      if (args.onProgressiveData && supportsMozChunked) {
//#endif
//#if !CHROMIUM
        xhr.responseType = 'moz-chunked-arraybuffer';
        pendingRequest.onProgressiveData = args.onProgressiveData;
        pendingRequest.mozChunked = true;
      }
//#endif
```

But in a [previous comment](https://github.com/mozilla/pdf.js/pull/5531/files#r33595781), you've expressed that you don't want this if-in-ifdef construct.

(The alternative (without boolean AND without if-in-ifdef) is duplication of code (copy & paste of the full `if` block, and include `supportsMozChunked` in non-FF/CR builds, and omit `supportsMozChunked` in FF builds). This is even uglier and undesirable.)

> In the future, I hope we can re-write this code and define better classes/interfaces than XHR that will be more suitable for our operations.

Definitely. The current code is too tightly coupled with XHR, and trying to replace it with the `fetch` API will be a challenge (in Chrome 43+, `fetch` supports readable streams, which enables PDF.js to display PDFs even if the server does not support range requests - #5319 & #6126).